### PR TITLE
Supporting `IN` and `NOT IN` as a filter type

### PR DESCRIPTION
**Pull request by sajjad-shirazy**
**My PR is a:**
- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix #issueNumber
- [ ] 💅 Enhancement
- [x] 🚀 New feature

**Main update on the:**
- [ ] Admin
- [ ] Documentation
- [x] Framework
- [ ] Plugin

**Manual testing done on the following databases:**
- [ ] Not applicable
- [x] MongoDB
- [x] MySQL
- [x] Postgres

**Description:**
Supporting `IN` and `NOT IN` as a filter, so we can have such queries:
```javascript
{
  books(limit: 10, where: {_id_nin: ["5c4dad1a8f3845222ca88a56", "5c4dad1a8f3845222ca88a57"]}) {
    _id,
    title
  }
}
```

### CODE REVIEW

**Comment by derrickmehaffy** [comment]
@sajjad-shirazy please add documentation and also please ensure you test with all 3 database types

**Comment by sajjad-shirazy** [comment]
ok

**Comment by sajjad-shirazy** [comment]
@derrickmehaffy done.

File: packages/strapi-hook-bookshelf/lib/index.js [hunk 1]
```diff
@@ -1107,7 +1107,14 @@ module.exports = function(strapi) {
           result.key = `where.${key}`;
           result.value = {
             symbol: 'IN',
-            value,
+            value: _.isArray(value) ? value : [value],
```
Comment (kamalbennani) [inline packages/strapi-hook-bookshelf/lib/index.js]: you can use _.castArray instead, it'll only cast the value to an array if it's not the case already

**Comment by kamalbennani** [comment]
I've already implemented this operator in this PR: https://github.com/strapi/strapi/pull/2452/files#diff-6bd63d5cf1dec74aafa2dd897ecfe932R23 but it may take some time to be merged/released

File: packages/strapi-hook-bookshelf/lib/index.js (hunk 1 above)
Comment (sajjad-shirazy) [reply packages/strapi-hook-bookshelf/lib/index.js]: Thanks, @kamalbennani I didn't know it.

File: packages/strapi-hook-mongoose/lib/index.js [hunk 2]
```diff
@@ -567,6 +567,10 @@ module.exports = function (strapi) {
           result.key = `where.${key}.$in`;
           result.value = value;
           break;
+        case '_nin':
+          result.key = `where.${key}.$nin`;
+          result.value = value;
```
[the remark this answers is not in the document]
Comment (sajjad-shirazy) [reply packages/strapi-hook-mongoose/lib/index.js]: You are right, I missed this case for mongoose.

**Pull request merged**